### PR TITLE
Use ClusterName as node-prefix

### PR DESCRIPTION
**Pull request by j-griffith**
This change removes the NodePrefix constants we were using (controlPlaneNode and
workerNode) and instead just uses the ClusterName provided by the spec.  So for
example, now you'll have:
  `<cluster-name>-master-1`
  `<cluster-name>-worker-1`
  `<cluster-name>-worker-2`

This means you can have multiple Rancher servers running on a vSphere (not
overly useful in the real world, but nice for development and testing purposes)

CI testing for example can use a shared vSphere and run multiple tests in
parallel.  A developer can have multiple versions and play around with each one
without tearing down and rebuilding.  It can also be used as a migration tool.

### CODE REVIEW

**Comment by j-griffith** [comment]
Still failing...

/hold

File: pkg/engine/rkecli/rkecli.go [hunk 1]
```diff
@@ -132,7 +132,8 @@ func (c *MgmtCluster) CreatePermanent() error {
 			Labels:           make(map[string]string),
 			Taints:           make([]rkeTaint, 0),
 		}
-		if strings.HasPrefix(strings.ToLower(k), "controlplane") {
+		masterPrefix := fmt.Sprintf("%s-master", c.ClusterName)
```
Comment (jacobweinstock) [inline pkg/engine/rkecli/rkecli.go:135]: i'd prefer if we didnt use the word master. could use something else instead? something like control or controlplane?

Comment (j-griffith) [reply pkg/engine/rkecli/rkecli.go:135]: Fine by me

File: pkg/engine/rkecli/rkecli.go [hunk 2]
```diff
@@ -132,7 +132,8 @@ func (c *MgmtCluster) CreatePermanent() error {
...
+		if strings.HasPrefix(strings.ToLower(k), masterPrefix) {
```
Comment (jacobweinstock) [inline pkg/engine/rkecli/rkecli.go:136]: i found another one of these that needs updating. might be time to make a constant out of this? 
https://github.com/j-griffith/cake/blob/use_clusterName_for_nodename_prefix/pkg/engine/rkecli/rkecli.go#L392

Comment (j-griffith) [reply pkg/engine/rkecli/rkecli.go:136]: > i found another one of these that needs updating. might be time to make a constant out of this?
> https://github.com/j-griffith/cake/blob/use_clusterName_for_nodename_prefix/pkg/engine/rkecli/rkecli.go#L392

Yeah, we have several of them scattered throughout, I'll snoop around for them all

File: pkg/provider/vsphere/rke.go [hunk 3]
```diff
@@ -8,6 +8,11 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
+const (
+	CONTROL_PLANE = "controlplane"
```
Comment (bgentry-ntap) [inline pkg/provider/vsphere/rke.go:12]: Might be a good candidate for the config package to be defined once and used all these places.

nit: const should be camelCase or CamelCase if moving it to a new pkg to export

Comment (j-griffith) [reply pkg/provider/vsphere/rke.go:12]: DOH, my bad.. this is missing the last commit in the chain.  Pushing the missing commit now

**Pull request merged**